datafusion/common/src/config.rs

```diff
@@ -2093,3 +2094,22 @@ mod tests {
         assert_eq!(parsed_metadata.get("key_dupe"), Some(&Some("B".into())));
     }
 }
+
+#[cfg(test)]
+mod tests_isolated {
```

Comment:
```suggestion
mod tests {
```

I think it may be best to stick to [convention](https://doc.rust-lang.org/book/ch11-03-test-organization.html#unit-tests) here.